drivers/bluetooth/hci/Kconfig

```diff
@@ -71,6 +71,7 @@ config BT_STM32WBA
 	bool "STM32WBA HCI driver"
 	default y
 	depends on DT_HAS_ST_HCI_STM32WBA_ENABLED
+	depends on ZEPHYR_HAL_STM32_MODULE_BLOBS
```

Comment:
i'm not sure this is a good idea.

A user who tries to build a sample that selects this driver (`CONFIG_BT_STM32WBA=y` will see the following message (or an even harder to understand one):
```
warning: BT_STM32WBA (defined at drivers/bluetooth/hci/Kconfig:70) was assigned the value 'y'
but got the value 'n'. Check these unsatisfied dependencies: ZEPHYR_HAL_STM32_MODULE_BLOBS (=n)
```

but how is user suppose to know that the way to solve this problem is to run `west blobs` ?

What is done else-where, is to print a more useful help text during CMake configure, like this:
https://github.com/zephyrproject-rtos/zephyr/blob/45705a84ab8ecbdf116fde47319f5cfd9f5890a9/modules/hal_infineon/cat1cm0p/CMakeLists.txt#L36
https://github.com/zephyrproject-rtos/zephyr/blob/45705a84ab8ecbdf116fde47319f5cfd9f5890a9/drivers/wifi/nrfwifi/CMakeLists.txt#L152

Reply:
@tejlmand hmm, I see your point. However, it then moves us back to square 1 with trying to address #76867. It's possible to build (but not run) the Zephyr Bluetooth host stack without an HCI driver, so with the `depends on` I would have been able to remove "bluetooth" from here: https://github.com/zephyrproject-rtos/zephyr/blob/143b14bb4fb779437fdeb8369e42fbb81db4a5fd/boards/silabs/dev_kits/xg24_dk2601b/xg24_dk2601b.yaml#L15-L18

The fact that we have to explicitly exclude certain tags right now (due to upstream CI trying to build and fail otherwise) combined with the fact that twister doesn't support overriding such filters, means that we can't use twister in downstream trees (where we can do `west blobs fetch` as part of CI, in contrast to upstream CI).

I agree that we need to find a solution which results with a sufficiently understandable error message for the user, but I'm also trying to work toward a solution where I can start removing these explicit testing tag exclusions in the board definitions.

Reply:
I also wanted to add that the reason I went with the current approach was after a lengthy discussion in #76867 which resulted in the following [recommendation from @nashif](https://github.com/zephyrproject-rtos/zephyr/issues/76867#issuecomment-2301923364) :

> IMO this could also be solved on the west/kconfig side of things, because the problem first appears when you try to build a sample with west/cmake for a board that needs some blob, for example for bluetooth. IMO this should be solved at this level.

Reply:
> A user who tries to build a sample that selects this driver (`CONFIG_BT_STM32WBA=y` will see the following message (or an even harder to understand one):
> ```
> warning: BT_STM32WBA (defined at drivers/bluetooth/hci/Kconfig:70) was assigned the value 'y'
> but got the value 'n'. Check these unsatisfied dependencies: ZEPHYR_HAL_STM32_MODULE_BLOBS (=n)
> ```
> but how is user suppose to know that the way to solve this problem is to run `west blobs` ?

It's at least not as bad as some common device tree build errors & warnings people get with Zephyr :)

One idea that comes to mind, is that we could keep the `depends on` but then check for the combination of the HCI device DT node being enabled but blobs not being present and then print an error message with the exact needed `west blobs fetch` command. This can't really be generalized though, i.e. it'd need to be done separately, either in each module's CMakeList.txt, or then for each necessary HCI driver in `drivers/bluetooth/hci/CMakeLists.txt`. The downside is that this is sort of duplication in addition to the `depends on`.

Reply:
> A user who tries to build a sample that selects this driver (CONFIG_BT_STM32WBA=y

Actually, these are not supposed to be user selectable. That's being fixed by #77418.

So in practice the knowledge of having to run `west blobs fetch ...` is similar to the knowledge that you need to know to set `status = "okay";` on the the right HCI devicetree node. That said, I still think it could be friendlier if in CMakeLists.txt we'd be checking for DT being enabled but blobs not available and then print a helpful message to the user.

Reply:
> It's possible to build (but not run) the Zephyr Bluetooth host stack without an HCI driver

I was wrong about this. I assumed that if the driver wasn't built, then the respective struct device instance would still exist (thanks to a node with `status = "okay";` in dts), and would make `device_is_ready()` return false, but actually this just causes a linking failure:
```
zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(hci_core.c.obj):(.data.bt_dev+0x16c): undefined reference to `__device_dts_ord_2'
```

This means that the new `depends on` for blobs doesn't seem to move us any closer to a solution for #76867.

Reply:
> What is done else-where, is to print a more useful help text during CMake configure, like this:

@tejlmand I added similar CMake warnings for each HCI driver. It might be possible to perform some deduplication by having a helper CMake function, but I left that as a potential follow-up exercise.